lib_xcore_c/api/xcore_c_chanend.h

```diff
@@ -0,0 +1,173 @@
+ *  \param func   The select_callback_t function to handle the event
+ *
+ *  \return     error_none (or exception type if policy is XCORE_C_NO_EXCEPTION).
+ *
+ *  \exception  ET_ILLEGAL_RESOURCE   not a valid chan-end.
+ *  \exception  ET_RESOURCE_DEP       another core is actively using the chan-end.
+ *  \exception  ET_ECALL              when xassert enabled, on XS1 bit 16 not set in enum_id.
+ */
+inline xcore_c_error_t chanend_setup_select_callback(chanend c, void *data, select_callback_t func)
+{
+  RETURN_EXCEPTION_OR_ERROR( _resource_setup_select_callback(c, data, func) );
+}
+
+/** Setup interrupt events on a chan-end .
+ *
+ *  Once the event is setup you need to call chanend_enable_trigger() to enable it.
+ *
+ *  \param c     The chan-end to setup the events on
+ *  \param data  The value to be passed to the interrupt_callback_t function
+ *  \param func  The interrupt_callback_t function to handle events
+ *
+ *  \return     error_none (or exception type if policy is XCORE_C_NO_EXCEPTION).
+ *
+ *  \exception  ET_ILLEGAL_RESOURCE   not a valid chan-end.
+ *  \exception  ET_RESOURCE_DEP       another core is actively using the chan-end.
+ *  \exception  ET_ECALL              when xassert enabled, on XS1 bit 16 not set in enum_id.
+ */
+inline xcore_c_error_t chanend_setup_interrupt_callback(chanend c, void *data,
+                                                 interrupt_callback_t func)
+{
+  RETURN_EXCEPTION_OR_ERROR( _resource_setup_interrupt_callback(c, data, func) );
+}
+
+/** Enable select & interrupt events on a chan-end.
+ *
+ *  Prior to enabling, chanend_setup_select(), chanend_setup_select_callback() or
+ *  chanend_setup_interrupt_callback() must have been called.
+ *  Events can be temporarily disabled and re-enabled using chanend_disable_trigger()
+ *  and chanend_enable_trigger().
+ *  When the event fires, the value must be read from the chan-end to clear the event.
+ *
+ *  \param c    The chan-end to enable events on
+ *
+ *  \return     error_none (or exception type if policy is XCORE_C_NO_EXCEPTION).
+ *
+ *  \exception  ET_ILLEGAL_RESOURCE   not a valid chan-end.
+ *  \exception  ET_RESOURCE_DEP       another core is actively using the chan-end.
+ */
+inline xcore_c_error_t chanend_enable_trigger(chanend c)
+{
+  RETURN_EXCEPTION_OR_ERROR( _resource_enable_trigger(c) );
+}
+
+/** Disable select & interrupt events for a given chan-end.
+ *
+ *  This function prevents any further events being triggered by a given chan-end.
+ *
+ *  \param c    The chan-end to disable events on
+ *
+ *  \return     error_none (or exception type if policy is XCORE_C_NO_EXCEPTION).
+ *
+ *  \exception  ET_ILLEGAL_RESOURCE   not a valid chan-end.
+ *  \exception  ET_RESOURCE_DEP       another core is actively using the chan-end.
+ */
+inline xcore_c_error_t chanend_disable_trigger(chanend c)
+{
+  RETURN_EXCEPTION_OR_ERROR( _resource_disable_trigger(c) );
+}
+
+#endif // __XC__
```

Comment:
Might avoid confusion for the "casual reader" (e.g. people that haven't studied the file from the top) if this comment was ``! __XC__``?